gateway/ln-gateway/src/rpc/rpc_client.rs

```diff
@@ -31,19 +32,26 @@ pub struct GatewayRpcClient {
     client: reqwest::Client,
     /// Optional gateway password
     password: Option<String>,
+    /// Optional gateway JWT
+    pub jwt_code: Option<String>,
 }
 
 impl GatewayRpcClient {
-    pub fn new(versioned_api: SafeUrl, password: Option<String>) -> Self {
+    pub fn new(versioned_api: SafeUrl, password: Option<String>, jwt_code: Option<String>) -> Self {
```

Comment:
What are the semantics of providing a password, a JWT or both? Please document. E.g. does providing the password lead to automatic requesting of a JWT whenever the previous expires?

Reply:
Providing both a password and a JWT does not make sense. 
Providing an initial password has the purpose of generating a JWT that has to be used to request private endpoints.

Maybe instead of providing optional parameters (JWT and password) would be better to provide an `enum` for each case,  WDYT?